Customer Demo/Customer Demo/Screens/Setup/SetupPresenter.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  SetupPresenter.swift
+//  Customer Demo
+//
+//  Copyright 2022 Ably Real-time Ltd (ably.com)
+//
+
+import UIKit
+import CoreLocation
+import AblyAssetTrackingSubscriber
+
+class SetupPresenter: NSObject {
+    
+    weak var viewController: SetupViewController?
+    
+    func handleSubmitAction(
+        sourceLocationText: (String, String),
+        destinationLocationText: (String, String),
+        minimumDisplacementText: String,
+        desiredIntervalText: String,
+        accuracyIndex: Int
+    ) {
+        if !validate(
+            sourceLocationText: sourceLocationText,
+            destinationLocationText: destinationLocationText,
+            minimumDisplacementText: minimumDisplacementText,
+            desiredIntervalText: desiredIntervalText,
+            accuracyIndex: accuracyIndex
+        ) {
+            return
+        }
+        
+        let sourceLocation = CLLocation(
+            latitude: Double(sourceLocationText.0) ?? 0,
+            longitude: Double(sourceLocationText.1) ?? 0
+        )
+        let destinationLocation = CLLocation(
+            latitude: Double(destinationLocationText.0) ?? 0,
+            longitude: Double(destinationLocationText.1) ?? 0
+        )
+        
+        createOrder(source: sourceLocation, destination: destinationLocation) { [weak self] response in
+            let storyboard = UIStoryboard(name: "DeliveryStatus", bundle: nil)
+            let deliveryStatusViewController = storyboard.instantiateInitialViewController() as? DeliveryStatusViewController
+            guard let deliveryStatusViewController = deliveryStatusViewController else {
+                fatalError()
+            }
+            
+            let accuracy = Accuracy(index: accuracyIndex) ?? .low
+            let minimumDisplacement = Double(minimumDisplacementText) ?? 0
+            let desiredInterval = Double(desiredIntervalText) ?? 0
+            
+            let resolution = Resolution(
+                accuracy: accuracy,
+                desiredInterval: desiredInterval,
+                minimumDisplacement: minimumDisplacement
+            )
+            
+            deliveryStatusViewController.configure(
+                resolution: resolution,
+                orderID: "\(response.orderID)",
+                jsonWebToken: response.ably.token
+            )
+
+            self?.viewController?.navigationController?.pushViewController(deliveryStatusViewController, animated: true)
+        }
+    }
+}
+
+private extension SetupPresenter {
+    
+    func createOrder(source: CLLocation, destination: CLLocation, _ completion: @escaping (OrderResponse) -> Void) {
```

Comment:
Hmm, maybe we should move the networking logic to a separate class/service? I'm not a biggest fan of putting it into a presenter.